tests/unit/xoopsLib/Xoops/Core/Kernel/XoopsModelFactoryTest.php

```diff
@@ -0,0 +1,46 @@
+<?php
+require_once(dirname(__FILE__).'/../../../../init_new.php');
+
+use Xoops\Core\Kernel\Handlers\XoopsBlockHandler;
+
+/**
+* PHPUnit special settings :
+* @backupGlobals disabled
+* @backupStaticAttributes disabled
+*/
+class xoopsmodelfactoryTest extends \PHPUnit_Framework_TestCase
+{
+    protected $myClass = 'Xoops\Core\Kernel\XoopsModelFactory';
+
+    public function setUp()
+    {
+    }
+
+    public function test_getInstance()
+    {
+        $class = $this->myClass;
+        $instance = $class::getInstance();
+        $this->assertInstanceOf($class, $instance);
+
+        $instance2 = $class::getInstance();
+        $this->assertSame($instance, $instance2);
+    }
+
+    public function test_loadHandler()
+    {
+        $handler = new XoopsBlockHandler();
+        $vars = array('one'=>1, 'two'=>2);
+
+        $class = $this->myClass;
+        $instance = $class::getInstance();
+        $this->assertInstanceOf($class, $instance);
+
+        $x = $instance->loadHandler($handler, 'read', $vars);
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Avoid variables with short names like $x. Configured minimum length is 3.](https://www.codacy.com/app/richard_3/XoopsCore/file/2442997768/issues/source?bid=3194562&fileBranchId=3245574#l38)